WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreSetupProgressView.swift

```diff
@@ -0,0 +1,90 @@
+import SwiftUI
+
+struct StoreSetupProgressView: View {
+    let isExpanded: Bool
+
+    let totalNumberOfTasks: Int
+
+    let numberOfTasksCompleted: Int
+
+    var body: some View {
+        HStack(alignment: .top) {
+            VStack(alignment: isExpanded ? .center : .leading, spacing: Layout.verticalSpacing) {
+                // Title label
+                Text(Localization.title)
+                    .fontWeight(.semibold)
+                    .if(isExpanded) { $0.titleStyle() }
+                    .if(!isExpanded) { $0.headlineStyle() }
+                    .multilineTextAlignment(isExpanded ? .center : .leading)
+
+                // Progress view
+                ProgressView(value: Double(numberOfTasksCompleted), total: Double(totalNumberOfTasks))
+                    .tint(.init(uiColor: .accent))
+                    .frame(width: isExpanded ? Layout.ProgressView.widthExpanded : Layout.ProgressView.widthCollapsed, height: Layout.ProgressView.height)
+
+                // Subtitle label
+                Text(String(format: isExpanded ? Localization.TasksCompleted.expanded : Localization.TasksCompleted.collapsed,
+                            numberOfTasksCompleted,
+                            totalNumberOfTasks))
+                    .footnoteStyle()
+                    .multilineTextAlignment(isExpanded ? .center : .leading)
+            }
+
+            Spacer()
+                .renderedIf(!isExpanded)
+
+            // More button
+            Button {
+                // TODO: Show the popup with feedback button
+            } label: {
+                Image(uiImage: .ellipsisImage)
+                    .flipsForRightToLeftLayoutDirection(true)
+                    .foregroundColor(Color(.textTertiary))
+            }
+            .renderedIf(!isExpanded)
+        }
+    }
+}
+
+private extension StoreSetupProgressView {
+    enum Layout {
+        static let horizontalSpacing: CGFloat = 16
+        static let verticalSpacing: CGFloat = 8
+
+        enum ProgressView {
+            static let height: CGFloat = 6
+            static let widthCollapsed: CGFloat = 152.5
+            static let widthExpanded: CGFloat = 205.8
```

Comment:
super nit: I was wondering if we can do a percentage-based layout instead of fixed width for all screen sizes, maybe with `GeometryReader`? from the design, it looks like the collapsed state has the progress bar width as 50% of the content width.

Reply:
I cannot align the view properly when I use `GeometryReader`. The view occupies all the available space instead of taking only what is needed. I am reverting [the commit](https://github.com/woocommerce/woocommerce-ios/pull/8948/commits/eb7a2221357407f162a304587eba2e7a4c940322) for now. I will try to revisit this in a future PR using preference keys.